chain/src/sync/compact_block.rs

```diff
@@ -18,18 +21,20 @@ use super::StatePayload;
 #[serde(try_from = "pb::CompactBlock", into = "pb::CompactBlock")]
 pub struct CompactBlock {
     pub height: u64,
-    // State payloads describing new state fragments.
+    /// State payloads describing new state fragments.
     pub state_payloads: Vec<StatePayload>,
-    // Nullifiers identifying spent notes.
+    /// Nullifiers identifying spent notes.
     pub nullifiers: Vec<Nullifier>,
-    // The block root of this block.
+    /// The block root of this block.
     pub block_root: block::Root,
-    // The epoch root of this epoch, if this block ends an epoch (`None` otherwise).
+    /// The epoch root of this epoch, if this block ends an epoch (`None` otherwise).
     pub epoch_root: Option<epoch::Root>,
-    // Latest FMD parameters. `None` if unchanged.
+    /// Latest FMD parameters. `None` if unchanged.
     pub fmd_parameters: Option<FmdParameters>,
-    // If the block indicated a proposal was being started.
+    /// If the block indicated a proposal was being started.
     pub proposal_started: bool,
+    /// Output prices for batch swaps occurring in this block.
+    pub swap_outputs: BTreeMap<TradingPair, BatchSwapOutputData>,
```

Comment:
This PR adds the batch swap outputs into the compact block, so that the view service can determine the corresponding notes purely from the contents of the `CompactBlock`. I think this should be a design rule going forward: all of a user's note state should be recoverable purely from the contents of the `CompactBlock`, even if the *interpretation* of that state, or further action on it, requires additional information (e.g., looking up asset IDs, or looking up validator rates to make a delegate/undelegate/etc).